### DISCUSSION TITLE
GitHub Classroom autograding broke on September 12, 2023

### DISCUSSION THREAD

**Discussion by kate-holdener** [community/community#67008] [GitHub Education]
### Hi everyone,

I have been using GitHub Classroom autograding feature, which was working very well. On September 12, I was adding tests for a new assignment and noticed that at some point during the day, the points associated with each test do not persist. Specifically, when I create an assignment, add a test, and assign points for passing the test, everything works well.

However, if I go to edit the assignment and edit a test, when I open the test UI, the points are not there. I'm attaching a screenshot from one of my assignments. The test in the screenshot had points associated with it. When I went to edit the test, the points were gone.

I think if you create tests and leave them alone (do not edit them), things work well. But as soon as you need to edit one test or add a new test, all points get removed.

<img width="1368" alt="Screen Shot 2023-09-13 at 9 41 30 AM" src="https://github.com/community/community/assets/42073871/d484a5c1-130e-46a2-8208-ea66373319a6">

**Comment by mh-skjelvareid** [comment]
I've had the same experience. Very frustrating! I'll have to release an assignment without points, but hopefully if this gets working again, it's possible to re-run all the autograding and get points for each student...?

**Reply by kate-holdener** [reply]
I found a workaround! If you create your tests through GitHub Classroom (forget the points), then accept the assignment yourself and clone the repo that got created, you'll get a .github directory in your repo. I copy that directory into the template I use for the assignment. From there, modify the template's .github/classroom/autograder.json to have non-null points. Since this is the template, when students accept the assignment based on this template, they'll get the autograder with the point values.

This workaround is nice, because your template now always has tests and points associated with them. I'll be doing this for all my templates. The only downside is when you download the grades from GitHub classroom, it will show that the score is out of 0. For example: 20/0. But I can live with that.

**Reply by kate-holdener** [reply]
Also, any time you change anything in the autograder through GitHub Classroom, all tests rerun. That's not the case with my template-based workaround.

**Reply by mh-skjelvareid** [reply]
Wow, thanks a lot for the tip! I'll give that a try. :)

**Comment by mh-skjelvareid** [comment]
I had a hard time finding out where to open a proper issue about this, so that it would reach one of the GitHub Classroom developers. I found the [education/autograding](https://github.com/education/autograding) repository today and opened an issue there: https://github.com/education/autograding/issues/190
